aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -0,0 +1,316 @@
+
+        Uses marshmallow schemas to validate if format specific attachment data
+        is valid for the specified message type. Only does structural and type
+        checks, does not validate if .e.g. the issuer value is valid.
+
+
+        Args:
+            message_type (str): The message type to validate the attachment data for.
+                Should be one of the message types as defined in message_types.py
+            attachment_data (Mapping): [description]
+                The attachment data to valide
+
+        Raises:
+            Exception: When the data is not valid.
+
+        """
+        mapping = {
+            PRES_20_REQUEST: DIFProofRequestSchema,
+            PRES_20_PROPOSAL: InputDescriptorsSchema,
+            PRES_20: VerifiablePresentationSchema,
+        }
+
+        # Get schema class
+        Schema = mapping[message_type]
+
+        # Validate, throw if not valid
+        Schema(unknown=RAISE).load(attachment_data)
+
+    def get_format_data(self, message_type: str, data: dict) -> PresFormatAttachment:
+        """Get presentation format and attachment objects for use in presentation ex messages.
+
+        Returns a tuple of both presentation format and attachment decorator for use
+        in presentation exchange messages. It looks up the correct format identifier and
+        encodes the data as a base64 attachment.
+
+        Args:
+            message_type (str): The message type for which to return the cred format.
+                Should be one of the message types defined in the message types file
+            data (dict): The data to include in the attach decorator
+
+        Returns:
+            PresFormatAttachment: Presentation format and attachment data objects
+
+        """
+        return (
+            V20PresFormat(
+                attach_id=format.api,
+                format_=ATTACHMENT_FORMAT[message_type][format.api],
+            ),
+            AttachDecorator.data_json(data, ident=format.api),
+        )
+    
+    async def create_exchange_for_proposal(
+        self,
+        pres_ex_record: V20PresExRecord,
+        pres_proposal_message: V20PresProposal,
+    ) -> None:
+        """Create a presentation exchange record for input presentation proposal."""
+
+    async def receive_pres_proposal(
+        self,
+        pres_ex_record: V20PresExRecord,
+        message: V20PresProposal,
+    ) -> None:
+        """Receive a presentation proposal from message in context on manager creation."""
+
+    async def create_exchange_for_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        pres_request_message: V20PresRequest,
+    ) -> None:
+        """Create a presentation exchange record for input presentation request."""
+
+    async def create_bound_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        name: str = None,
+        version: str = None,
+        nonce: str = None,
+        comment: str = None,
```

Comment:
I think we should make this an object (e.g. `request_data`) where we expect certain things in the indy handler and other things in the PE handler. As those values are all indy specific and unused in this handler